File: leetcode/1111/2.js
var maxDepthAfterSplit = function(seq) {
  let dep = 0
  return seq.split('').map(item => item === '(' ? dep++%2 : --dep%2)
}


var seq = "(()())"
var seq ="(((()))((())))" 
var seq ="((()()))"
console.log(maxDepthAfterSplit(seq))

File: leetcode/1111/2.go
func maxDepthAfterSplit(seq string) []int {
	stack := make([]int, 0, len(seq))
	var deep = 0
	for _,v := range seq {
		if string(v) == "(" {
			stack = append(stack, deep%2)
			deep++
		} else {
			deep--
			stack = append(stack, deep%2)
		}
	}
	return stack
}